### ISSUE TITLE
Hope this can support Julia

### ISSUE THREAD

**Issue by carlparte** [DonJayamanne/vscodeJupyter#81]
Now I am using Julia and find that Jupyter is really a good tool. So hope you can add Julia kernel so that I can use this in vscode.

**Comment by DonJayamanne** [comment]
I haven't tested this myself, but it should work (theoretically).
I know it work for JavaScript fairly easily.

Please install the julia kernel and use the commands to send the selected text to a kernel.
Will look into this from my end too.

**Comment by findmyway** [comment]
I have tried Julia, most work fine. 

A little problem is that **Run Cell** doesn't show up.
Maybe support from https://github.com/JuliaEditorSupport/julia-vscode is needed

**Comment by carlparte** [comment]
@findmyway 
That's what I want to say！ The **Run Cell** option is really needed so that I can work within VS Code.

**Comment by malmaud** [comment]
Do you guys know about https://github.com/JuliaEditorSupport/julia-vscode? It's a pretty mature package that lets you evaluate Julia codeblocks like Jupyter, while also offering a lot of  Julia intellisense features.

**Comment by davidanthoff** [comment]
It is not clear to me what support we (the julia-vscode devs) would provide for the scenario here, I think the two projects just work independently?

**Comment by findmyway** [comment]
@malmaud 
Yes, I use  julia-vscode for my daily work. It's pretty awesome! 

However, julia-vscode doesn't support remote REPL(like REPL in Clojure).  With vscodeJupyter, I can start Jupyter notebook on a remote server and then execute commands on a local machine.

@davidanthoff 
Currently, the vscodeJupyter works fine to send commands and receive result. The only inconvenience we discussed here is that, Jupyter provides the concept of ``Cell``(a code block), so that we can execute them once at all. 
I think this feature is also needed in julia-vscode, I usually have to select multiple lines(a code block logically) and then send them to REPL back and forth.

The **Run Cell** button is shown as follows:

If a comment line start with ``#%%``, then the following code block will be considered as a **Cell**. Click **Run Cell** will send the following code to notebook server

![image](https://user-images.githubusercontent.com/5612003/35897742-a40a15da-0bfd-11e8-9d6c-25f954ef53c3.png)

Since I'm new to Julia, I'm not very sure if this feature is useful to others. Personally I would like to have this feature, so I can bind the **Run Cell** command to a short-cut. It would be really convenient.

Thanks again for your guys' efforts!

**Comment by davidanthoff** [comment]
Ah, I see, that is cool! Could you open an issue over on the julia extension repo with this idea?

**Comment by DonJayamanne** [comment]
Hi everyone, you should be able to define the cell identifier as follows:
* Go into your user settings file (`Preferences -> Settings`)
* Add the setting for `"Jupyter.languages"`
* As the following Jon 
```json
"Jupyter.languages": [
            {
              "languageId": "python",
              "startupCode": [
                "%matplotlib inline"
              ],
              "cellIdentificationPattern": "^(#\\s*%%|#\\s*\\<codecell\\>|#\\s*In\\[\\d*?\\]|#\\s*In\\[ \\])(.*)"
            },
            {
              "languageId": "javascript",
              "cellIdentificationPattern": "^(\\/\\/ %%|\\/\\/%%|\\/\\/ \\<codecell\\>|\\/\\/ In\\[\\d*?\\]|\\/\\/ In\\[ \\])(.*)"
            }
          ]
```

As another issues into the array for Julia with regular expression appropriately set to identify cells.

Of course other extensions can contribute to thid extension by using the API exposed by the Jupyter extension to provide the code lenses as well.
